### ISSUE TITLE
Simplifying h_outflow calculation when allowFlowReversal = false

### ISSUE THREAD

**Issue by Mathadon** [iea-annex60/modelica-annex60#281]
Currently Dymola is able to simplify some calculations when ```allowFlowReversal = false```. Consider for instance the following equation.
```H_in = semiLinear(m_flow, inStream(port_a.h_outflow), inStream(port_b.h_outflow))``` 
```allowFlowReversal``` typically sets the ```min``` attribute of ```m_flow``` equal to zero. Dymola can then detect that mass flow will never become negative and that therefore ```inStream(port_a.h_outflow)``` should always be used in the equation: 
```H_in = m_flow*inStream(port_a.h_outflow)```
Since the equation now no longer depends on  ```inStream(port_b.h_outflow)``` this can lead to smaller algebraic loops.

However there are some cases where the ```min``` and ```max``` attributes are not propagated properly by Dymola and therefore these simplifications are not made. Moreover OpenModelica does not make these simplifications at all.  This can lead to large algebraic loops, which deteriorate simulation speed and stability.

I propose to make some changes in the code such that this simplification is always made. To this end I would like to change ```Annex60.Fluid.Interfaces.StaticTwoPortConservationEquation```, ```Buildings.Fluid.BaseClasses.PartialResistance```, ```Annex60.Fluid.Sensors.BaseClasses.PartialFlowSensor``` and similar models. 

In the case of ```Annex60.Fluid.Sensors.BaseClasses.PartialFlowSensor``` the change would be to replace

```
  port_a.h_outflow = inStream(port_b.h_outflow);
  port_b.h_outflow = inStream(port_a.h_outflow);
  port_a.Xi_outflow = inStream(port_b.Xi_outflow);
  port_b.Xi_outflow = inStream(port_a.Xi_outflow);
  port_a.C_outflow = inStream(port_b.C_outflow);
  port_b.C_outflow = inStream(port_a.C_outflow);
```

by
```
  port_a.h_outflow = if not allowFlowReversal then Medium.h_default else inStream(port_b.h_outflow);
  port_b.h_outflow =  inStream(port_a.h_outflow);
  port_a.Xi_outflow = if not allowFlowReversal then Medium.Xi_default else  inStream(port_b.Xi_outflow);
  port_b.Xi_outflow = inStream(port_a.Xi_outflow);
  port_a.C_outflow = if not allowFlowReversal then Medium.C_default else  inStream(port_b.C_outflow);
  port_b.C_outflow = inStream(port_a.C_outflow);
```

It may actually be interesting the create a new partial that contains these equations since they are now repeated in a few partials.

I will add an example that demonstrates that this works.

**Comment by Mathadon** [comment]
See example  ```Annex60.Fluid.Interfaces.Examples.ReverseFlowAdvanced``` on https://github.com/Mathadon/modelica-annex60/tree/issue281_floRevCalc
Heat is injected based on the _upstream_ temperature of the fluid. Since mass flow is always in the same direction this temperature is known a priori. However since dymola thinks the flow can reverse (even with allowFlowReversal=false) a non-linear system is formed:

```
Sizes of nonlinear systems of equations: {3}
Sizes after manipulation of the nonlinear systems: {1}
```

This can be removed by setting 
```
port_a.h_outflow = if allowFlowReversal then Medium.h_default else inStream(port_b.h_outflow);
```
in ```Annex60.Fluid.Sensors.BaseClasses.PartialFlowSensor```.

**Comment by Mathadon** [comment]
If you agree with the proposed changes I can propose an implementation.

**Comment by Mathadon** [comment]
Illustration:
![screen shot 2015-06-30 at 11 24 18](https://cloud.githubusercontent.com/assets/5970879/8427820/9aea19d2-1f1a-11e5-9578-2933ea888b24.png)

**Comment by Mathadon** [comment]
I need to make a new pull request for this.

**Comment by rfranke** [comment]
Note that OpenModelica now considers the simplification. In fact any Modelica tool must do so in order to produce correct results for fluid models with absolute sensors, like Modelica.Fluid.Examples.{DrumBoiler, TraceSubstances.RoomCO2}.
See: https://trac.openmodelica.org/OpenModelica/ticket/3430

**Comment by Mathadon** [comment]
Merged in #312
